test/unit/foundation/schema/attribute.test.ts

```diff
@@ -0,0 +1,386 @@
+import { expect } from '@open-wc/testing';
+
+import { getSchema } from '../../../../src/schemas.js';
+import { schemaAttributeDefinition } from '../../../../src/foundation.js';
+
+const schema2007B4: XMLDocument = <XMLDocument>(
+  new DOMParser().parseFromString(
+    getSchema('2007', 'B', '4'),
+    'application/xml'
+  )
+);
+
+const schema2007B: XMLDocument = <XMLDocument>(
+  new DOMParser().parseFromString(getSchema('2007', 'B', ''), 'application/xml')
+);
+
+const schema2003: XMLDocument = <XMLDocument>(
+  new DOMParser().parseFromString(getSchema('', '', ''), 'application/xml')
+);
+
+describe('schema spcific functions', () => {
+  describe('allow to get attribute definition from schema that', () => {
+    it('returns empty array for invalid tagName', () =>
+      expect(schemaAttributeDefinition('nonsence', schema2007B4)).to.be.empty);
+
+    describe('for element SampledValueControl', () => {
+      describe('and version 2003', () => {
+        const attributes = schemaAttributeDefinition(
+          'SampledValueControl',
+          schema2003
+        );
+
+        it('return the correct number of attributes ', () =>
+          expect(attributes).to.have.length(8));
+
+        it('return an optional pattern', () =>
+          expect(
+            attributes.find(attribute => attribute.name === 'name')?.restriction
+              .base
+          ).to.equal('xs:normalizedString'));
+
+        it('return an optional pattern', () =>
+          expect(
+            attributes.find(attribute => attribute.name === 'name')?.restriction
+              .pattern
+          ).to.be.undefined);
+
+        it('return an optional minLength', () =>
+          expect(
+            attributes.find(attribute => attribute.name === 'name')?.restriction
+              .minLength
+          ).to.equal('1'));
+
+        it('return an otional maxLength', () =>
```

Comment:
```suggestion
        it('return an optional maxLength', () =>
```

And a few with this typo